pkg/utils/helper.go

```diff
@@ -11,6 +12,64 @@ import (
 	"strings"
 )
 
+// PublicAzureStorageEndpointSuffix defines default Storage Endpoint Suffix
+const PublicAzureStorageEndpointSuffix = "core.windows.net"
+
+// Azure defines Azure configuration
+type Azure struct {
+	Cloud string `json:"cloud"`
+}
+
+// AzureStackCloud defines Azure Stack Cloud configuration
+type AzureStackCloud struct {
+	StorageEndpointSuffix string `json:"storageEndpointSuffix"`
+}
+
+// IsAzureStackCloud returns true if the application is running on Azure Stack Cloud
+func IsAzureStackCloud() bool {
+	azureFile, err := RunCommandOnHost("cat", "/etc/kubernetes/azure.json")
+	if err != nil {
+		return false
+	}
+	var azure Azure
+	err = json.Unmarshal([]byte(azureFile), &azure)
+	if err != nil {
+		return false
+	}
+	cloud := azure.Cloud
+	return strings.ToUpper(cloud) == "AZURESTACKCLOUD"

```

Comment:
💡 is this `AZURESTACKCLOUD` well known constant, can you please add a doc which has this reference as comment above please, (For case of reference)

Reply:
I don't believe there is any documentation on this, "AzureStackCloud" is just the string that appears in the json file under that key if the application is running in Azure Stack. I will make it a string contact and use that reference instead and add a comment.

Reply:
https://kubernetes-sigs.github.io/cloud-provider-azure/install/configs/#azure-stack-configuration

Reply:
thanks @jadarsie for the documentation -- looks like `AzureStackCloud `is a well known constant.